config/initializers/content_security_policy.rb

```diff
@@ -7,10 +7,18 @@
 Rails.application.config.content_security_policy do |policy|
   policy.default_src :self
   policy.font_src    :self, 'https://www2.buildkiteassets.com/'
-  policy.img_src     :self, 'https://buildkiteassets.com/', 'https://buildkite.com/', Matomo::URL
+  policy.img_src     :self, 'https://buildkiteassets.com/', 'https://buildkite.com/', Matomo::URL, :data, :https, :http
   policy.object_src  :none
   policy.script_src  :self, Matomo::URL
-  policy.style_src   :self, :unsafe_inline
+  policy.script_src  :self, :unsafe_inline, :strict_dynamic, :report_sample, :https, :http
+
+  # Allow unsafe_eval in development. Mostly because we do the same in the buildktie app, but also
+  # because it's required to load the algolia docsearch JS
+  if Rails.env.development?
+    #policy.script_src :self, :unsafe_eval, Matomo::URL, :unsafe_inline, :strict_dynamic, :report_sample, :https, :http
+    policy.script_src :unsafe_eval, :unsafe_inline, :strict_dynamic, :report_sample, :https, :http
+  end
+  policy.connect_src 'https://*.algolia.net', 'https://*.algolianet.com'
```

Comment:
I assume my changes here are wrong 🙈

Reply:
That looks okay to me? Anything that you’re particularly worried about?

Reply:
I’m spin this up and see if we can find a way to have a stricter CSP, given how tiny the docs app is.

Reply:
> Anything that you’re particularly worried about?

Just that I don't understand CSP. I can see a lot of care went into crafting the CSP rules on buildkite.com, and the rules that I cargo-culted into here had very little care put into them.

> I’m spin this up and see if we can find a way to have a stricter CSP

I'm not sure 100% confident what makes a CSP more or less strict. Would you have time for a quick pairing session on this?

Reply:
Yep, sounds good @yob!